wgsl/index.bs

```diff
@@ -989,86 +988,365 @@ Host-shareable type |S| satisfies <dfn noexport>uniform buffer layout</dfn> when
+
 <table class='data'>
   <thead>
     <tr><th>Type<th>Description
   </thead>
-  <tr><td>ptr<*SC*,*T*><td>Pointer (or reference) to storage in [=storage class=] *SC*
-                            which can hold a value of the [=storable=] *T*.
-                            Here, *T* is the known as the *pointee* type.
+  <tr algorithm="pointer type">
+    <td>ptr&lt;|SC|,|T|&gt;
+    <td>Pointer (or reference) to storage in [=storage class=] |SC|
+        which can hold a value of the [=storable=] type |T|.
+        Here, |T| is the known as the <dfn noexport>pointee type</dfn>.
 </table>
 
-Note: We've described a SPIR-V logical pointer type.
-
-Note: Pointers are not storable.
-
-<div class='example wgsl' heading='Pointer'>
+<div class='example wgsl' heading='Pointer type (program fragment)'>
   <xmp highlight='rust'>
-    ptr<storage, i32>
-    ptr<private, array<i32, 12>>
+    // The type of of a pointer value that references storage for keeping
+    // an 'i32' value, using memory locations in the 'uniform' storage class.
+    // Here 'i32' is the pointee type.
+    ptr<uniform, i32>              // Note: This is not a valid expression on its own.
+
+    // The type of a pointer value that references storage for keeping an array
+    // of 50 elements of type 'f32', using memory locations in the 'private'
+    // storage class.
+    // Here the pointee type is 'array<f32,50>'.
+    ptr<private, array<f32, 50>>   // Note: This is not a valid expression on its own.
   </xmp>
 </div>
 
-### Abstract Operations on Pointers TODO ### {#abstract-pointer-operations}
-
-A pointer value *P* supports the following operations:
-
-<table class='data'>
-  <tr><td>P.Write(V)<td>Place a value V into the referenced storage.
-               V’s type must match P’s pointee type.
-  <tr><td>P.Read()<td>An evaluation yielding the value currently in the P’s
-             referenced storage.  The result type is P's pointee type.
-  <tr><td>P.Subaccess(K)<td>Valid for pointers with a composite pointee type where
-                   *K* must evaluate to an integer between 0 and one
-                   less than the number of components in *P*’s pointee type.
-                   The subaccess evaluation yields a pointer to the storage for
-                   the K’th component within P’s referenced storage,
-                   using zero-based indexing. If P's [=storage class=] is SC, and
-                   the K'th member of P's pointee type is of type T, then
-                   the result type is `ptr<SC,T>`.
-</table>
+### Forming a pointer value ### {#forming-a-pointer-value}
+
+When evaluating an expression, a pointer value is formed in one of the
+following ways:
+
+* By using the name of a variable that is in scope.
+    The result is the pointer to the storage allocated for the variable.
+* By a <dfn noexport>composite value pointer sub-access</dfn> expression:
+    * Given a pointer to a vector, appending a single-letter vector access phrase
```

Comment:
How does that work with the pointer? I'm saying:

```
var a : vec3<f32>;
const b : ptr<function, vec2<f32>> = a.xz;
```

would be a validation error, yes? Or, are you saying that `a.xz` loads into a temporary and we take the address of the temporary?

Reply:
Yes, that's a validation error.

Reply:
It's a validation error because you can't form a pointer type from the right hand side. 
You would *like* to use a rule from section "Component pointer from vector pointer"  But that only has rules for a single letter after the dot.

Reply:
Right, I think we should add a brief 'multi-letter swizzles are a validation error' just to make that clear in the doc.

Reply:
By the way, in Metal, you can't take a pointer to a member of a vector, even if that member is a single field.